source/includes/fundamentals/code-snippets/MonolightCodecProvider.java

```diff
@@ -0,0 +1,35 @@
+package fundamentals.monolightcodec;
+
+import org.bson.codecs.Codec;
+import org.bson.codecs.IntegerCodec;
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistry;
+
+
+// start class
+public class MonolightCodecProvider implements CodecProvider {
+    private MonolightCodec monolightCodec;
+    private IntegerCodec integerCodec;
+    private PowerStatusCodec powerStatusCodec;
+
+    public MonolightCodecProvider() {
+        integerCodec = new IntegerCodec();
+        powerStatusCodec = new PowerStatusCodec();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> Codec<T> get(Class<T> clazz, CodecRegistry registry) {
+        if (clazz == Monolight.class) {
+            return (Codec<T>) new MonolightCodec(registry);
+        } else if (clazz == Integer.class) {
```

Comment:
Its an anti pattern for a provider to handle multiple types. Especially, if they have their own codecs.  In this scenario you would use `fromCodecs(new IntegerCodec(), new PowerStatusCodec());`

Reply:
Do you mean I should be instantiating a new CodecRegistry (using the factory method you provided) and then fetching the `IntegerCodec` or `PowerStatusCodec` from it?

Reply:
Not quite, its just that a custom CodecProvider typically would only handle a single type and not multiple types as shown here.